jmbo/models.py

```diff
@@ -243,6 +243,34 @@ def get_absolute_url(self):
             # Fallback
             return reverse('object_detail', args=[self.slug])
 
+    def get_absolute_url_categorized(self):
+        """Absolute url with category.
+
+        Provides a hook to get an url for an object, connected to a category,
+        but just reusing the get_absolute_url templates etc. This differs from
+        the get_absolute_category_url method in that we don't provide
+        different sets of templates.
+        """
+        if self.primary_category:
+            category_slug = self.primary_category.slug
+        elif self.categories.all().exists():
+            category_slug = self.categories.all()[0].slug
+
+        if category_slug:
```

Comment:
category_slug can be undefined. You're missing a category_slug = None at the top of the method.